src/webgpu/gpu_test.ts

```diff
@@ -684,6 +685,110 @@ export class GPUTest extends Fixture<GPUTestSubcaseBatchState> {
     );
   }
 
+  /**
+   * Emulate a texture to buffer copy by using a compute shader
+   * to load texture value of a single pixel and write to a storage buffer.
+   * For sample count == 1, the buffer contains only one value of the sample.
+   * For sample count > 1, the buffer contains (N = 2 ** sampleCount) values sorted
+   * in the order of their sample index [0, 2 ** sampleCount - 1]
+   *
+   * This can be useful when the texture to buffer copy is not available to the texture format
+   * e.g. (depth24plus)
+   *
+   * MAINTENANCE_TODO: extend to read multiple pixels with given origin and size.
+   *
+   * @returns storage buffer containing the copied value from the texture.
+   */
+  copySinglePixelTextureToBufferUsingComputePass(
+    type: ScalarType,
+    numElements: number,
+    textureView: GPUTextureView,
+    sampleCount: number
+  ): GPUBuffer {
+    const textureLoadCode =
+      numElements === 1
+        ? `dst.data[sampleIndex] = textureLoad(src, coord, sampleIndex).x;`
+        : `let o = sampleIndex * ${numElements};
+      let v = textureLoad(src, coord, sampleIndex);
+      dst.data[o] = v.r;
+      dst.data[o + 1] = v.g;
+      dst.data[o + 2] = v.b;
+      dst.data[o + 3] = v.a;
```

Comment:
I think this code is assuming 4 components. I think you can write a loop that works regardless of the number of components? (and put inline in the shader below)

```
let o = sampleIndex * ${numElements};
let v = textureLoad(src, coord, sampleIndex);
for (var component = 0; component < ${componentCount}; component = component + 1) {
  dst.data[o + component] = v[component];
}
```